internal/app/synchronizer/evaluator_client.go

```diff
@@ -119,10 +119,15 @@ func (ec *grcpEvaluatorClient) evaluate(ctx context.Context, pc <-chan []*pb.Mat
 	}
 
 	results := []string{}
+	matchIDs := map[string]bool{}
```

Comment:
actually, I just noticed this is being used by multiple go routines at the same time.  Needs to be a sync.map